File: core/src/test/java/javabot/model/TestUser.java
package javabot.model;

import javabot.Messages;
import org.pircbotx.PircBotX;
import org.pircbotx.User;
import org.pircbotx.UserChannelDao;
import org.pircbotx.output.OutputUser;

import javax.inject.Inject;

public class TestUser extends User {
    private final PircBotX ircBot;
    private Messages messages;

    public TestUser(final PircBotX ircBot,
                    final UserChannelDao userChannelDao,
                    final Messages messages,
                    final String nick,
                    final String login,
                    final String host) {
        super(ircBot, userChannelDao, nick);
        this.ircBot = ircBot;
        this.messages = messages;
        setHostmask(host);
        setLogin(login);
    }

    @Override
    public OutputUser send() {
        return new OutputUser(ircBot, TestUser.this) {
            @Override
            public void message(final String message) {
                messages.add(message);
            }
        };
    }

    @Override
    public String toString() {
        return getNick();
    }
}

File: src/test/kotlin/javabot/model/TestUser.kt
package javabot.model

import javabot.Messages
import javabot.MockUserHostmask
import org.pircbotx.PircBotX
import org.pircbotx.User
import org.pircbotx.output.OutputUser

class TestUser(
    private val ircBot: PircBotX,
    private val messages: Messages,
    nick: String,
    login: String,
    host: String
) : User(MockUserHostmask(ircBot, nick, login, host)) {

    override fun send(): OutputUser {
        return object : OutputUser(ircBot, this@TestUser) {
            override fun message(message: String) {
                messages.add(message)
            }
        }
    }
}
